cmd/image-processing/main_test.go

```diff
@@ -193,177 +166,132 @@ var _ = Describe("Image Processing Resource", func() {
 
 		It("should fail in case --image does not exist", func() {
 			Expect(run(
-				"--image",
-				"docker.io/feqlQoDIHc/bcfHFHHXYF",
+				"--image", "docker.io/feqlQoDIHc/bcfHFHHXYF",
 			)).To(HaveOccurred())
 		})
 
 		It("should fail in case annotation is invalid", func() {
-			tag := pushImage("test1")
-
-			Expect(run(
-				"--image",
-				tag.String(),
-				"--annotation",
-				"org.opencontainers.image.url*https://my-company.com/images",
-			)).To(HaveOccurred())
+			withTestImage(func(tag name.Tag) {
```

Comment:
Instead of `pushImage` (that was never cleaned-up) as a standalone call, the pattern would here be that you have `withTestImage` and when it leaves the provided inner function, the clean-up is done automatically.